common/unsafe/src/main/java/org/apache/spark/unsafe/memory/UnsafeMemoryAllocator.java

```diff
@@ -19,28 +19,61 @@
 
 import org.apache.spark.unsafe.Platform;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+
 /**
  * A simple {@link MemoryAllocator} that uses {@code Unsafe} to allocate off-heap memory.
  */
 public class UnsafeMemoryAllocator implements MemoryAllocator {
 
+  private static Method bufAddrMethod;
+  static {
+    try {
+      Class cb = UnsafeMemoryAllocator.class.getClassLoader().loadClass("java.nio.DirectByteBuffer");
+      bufAddrMethod = cb.getMethod("address");
```

Comment:
You could use `sun.nio.ch.DirectBuffer` here. Both are evil.

Reply:
Thanks, I used this class.